### ISSUE TITLE
Using typeshed stubs instead of dependencies 

### ISSUE THREAD

**Issue by xrmx** [pylint-dev/pylint#9574]
### Question

Would it be possible (or even make sense) to use to use the stubs from typeshed instead of the real libraries in pylint checks?

```
import requests

requests.get("https:")
```

returns

```
foo.py:3:0: W3101: Missing timeout argument for method 'requests.get' can cause your program to hang indefinitely (missing-timeout)
```

with requests installed but nothing with just types-requests.

### Documentation for future user

No idea, it would be nice to have an example in the tutorial using a third party library and explicitly saying you need to install it in order to provide more checks.

**Comment by Pierre-Sassoulas** [comment]
Thank you for the suggestion. I added the need design proposal because I don't know if we should always use the stub if it exists or leave the choice to the user, or how to remove the lib files from the analysis if the stub exists. It feels non trivial :)

**Comment by nickserv** [comment]
This would be very helpful. I've noticed false positives where Pylint thinks a usage of a type from Typeshed is incorrect based on the real module. For example, `unsubscriptable-object` has a false positive with `networkx.Graph[T]`, since `__class_getitem__` is missing from `networkx.Graph` and the subscript is only used for its generic type in Typeshed.
